smoke_test/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+# First copy the CMakeList.txt so we can build the test
+file(COPY ${PACKAGE_SOURCE_DIR}/smoke_test/CMakeLists.txt.in DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/source)
+file(RENAME ${CMAKE_CURRENT_BINARY_DIR}/source/CMakeLists.txt.in ${CMAKE_CURRENT_BINARY_DIR}/source/CMakeLists.txt)
```

Comment:
We could probably do this in a single `configure_file` command, which would have better dependency tracking behavior anyway